templates/deployment-analyzer.yaml

```diff
@@ -110,16 +129,91 @@ spec:
       {{- if .Values.env }}
 {{ toYaml .Values.env | indent 8 }}
       {{- end }}
+{{- if .Values.exporter.enabled }}
+{{ toYaml .Values.exporter.env | indent 8 }}
+{{- end }}
         resources:
 {{ toYaml .Values.resources | indent 10 }}
         volumeMounts:
         - name: ssl
           mountPath: /etc/ssl/certs
+{{- if .Values.exporter.enabled }}
+      - name: skydive-exporter
+        image: {{ .Values.exporter.image.repository }}:{{ .Values.exporter.image.tag }}
+        securityContext:
+          privileged: false
+          allowPrivilegeEscalation: false
+          runAsNonRoot: true
+          runAsUser: 1000
+        env:
+        - name: SKYDIVE_ANALYZERS
+          value: {{ template "fullname" . }}-service:{{ .Values.service.port }}
+        - name: SKYDIVE_PIPELINE_SUBSCRIBER_URL
+          value: ws://{{ template "fullname" . }}-service:{{ .Values.service.port }}/ws/subscriber/flow
+        - name: SKYDIVE_PIPELINE_STORE_BUFFERED_FILENAME_PREFIX
+          valueFrom:
+            configMapKeyRef:
+              name: skydive-exporter-s3-configuration
+              key: "objectPrefix"
+        - name: SKYDIVE_PIPELINE_WRITE_S3_ENDPOINT
+          valueFrom: 
+            configMapKeyRef:
+               name: "skydive-exporter-s3-configuration"
+               key: "endpoint"
+        - name: SKYDIVE_PIPELINE_WRITE_S3_ACCESS_KEY
+          valueFrom:
+            secretKeyRef:
+              name: "skydive-exporter-secret"
+              key: "accesskey"
+        - name: SKYDIVE_PIPELINE_WRITE_S3_SECRET_KEY
+          valueFrom:
+            secretKeyRef:
+              name: "skydive-exporter-secret"
+              key: "secretkey"
+      {{- if .Values.exporter.write.s3.use_api_key }}
+        - name: SKYDIVE_PIPELINE_WRITE_S3_API_KEY
+          valueFrom:
+            secretKeyRef:
+              name: "skydive-exporter-secret"
+              key: "apikey"
+      {{- end }}
+        - name: SKYDIVE_PIPELINE_CLASSIFY_CLUSTER_NET_MASKS
+          valueFrom:
+            configMapKeyRef:
+              name: skydive-configuration
+              key: "netmasks"
+        - name: SKYDIVE_PIPELINE_STORE_BUFFERED_DIRNAME
+          valueFrom:
+            configMapKeyRef:
+              name: "skydive-exporter-s3-configuration"
+              key: "bucket"
+        - name: SKYDIVE_PIPELINE_WRITE_S3_REGION
+          valueFrom:
+            configMapKeyRef:
+              name: "skydive-exporter-s3-configuration"
+              key: "region"
+      {{- if .Values.exporter.env }}
+{{ toYaml .Values.exporter.env | indent 8 }}
+      {{- end }}
+{{- end }}
     {{- if .Values.persistence.enabled }}
       - name: skydive-elasticsearch
         image: {{ .Values.elasticsearch.image.repository }}:{{ .Values.elasticsearch.image.tag }}
         ports:
         - containerPort: {{ .Values.storage.elasticsearch.port }}
+        securityContext:
+          privileged: false
+          readOnlyRootFilesystem: false
```

Comment:
I think we might be able to run the flow-exporter as `readOnlyRootFilesystem: true` .

Reply:
from line 199 on the elasticsearch container is described, so it is not something related to exporter and was added by mistake. Removing the securityContext entirely